components/director/internal/open_resource_discovery/service.go

```diff
@@ -57,12 +58,22 @@ func NewAggregatorService(transact persistence.Transactioner, labelRepo labelRep
 		vendorSvc:          vendorSvc,
 		tombstoneSvc:       tombstoneSvc,
 		tenantSvc:          tenantSvc,
+		globalRegistrySvc:  globalRegistrySvc,
 		ordClient:          client,
 	}
 }
 
 // SyncORDDocuments performs resync of ORD information provided via ORD documents for each application
 func (s *Service) SyncORDDocuments(ctx context.Context) error {
+	globalResourcesOrdIDs, err := s.globalRegistrySvc.SyncGlobalResources(ctx)
```

Comment:
Just thoughts - adding this sync here makes the global resources sync a hard dependency in order to process any other app ord documents. From one PoV it makes sense since the app ord docs can reference new things coming from the glob registry and we need to make sure we have them. From another PoV the global registry probably won't change much and even when it does, we'll most likely fetch the new changes in it well before any app ord doc has adopted the new product/vendor changes which might have appeared in the global registry.

I'm saying all this just for us to consider if having this sync as hard dependency can backfire in situations where it could go missing and it would block us from syncing and new app ord documents which might have appeared along the way.
I'm okay with leaving it like that tbh, but we need to be aware of the possible problem it could cause and be ready to change the global resource sync to an independent job if it happens.

Reply:
This is the reason I made it optional by just logging if something goes wrong instead of failing the whole aggregator execution. However, you are right that some app syncs will fail because we will proceed with empty globalResourcesOrdIDs.

What do you think about the approach where in the error case (when global registry sync fails) we not only log but get all globalResourcesOrdIDs from the database (that were available in the global registry on the last successful execution)? This way we can safely process apps that reference global resources we already know and once the registry is up again we will resync them again.

Reply:
Wow, i totally missed that you log only - I thought i saw that it errored.
But generally, yes, I think it would be best to fetch the global resources because otherwise it would almost definitely fail, because our current lob always uses a global resource